freeipa/src/main/java/com/sequenceiq/freeipa/util/HealthCheckAvailabilityChecker.java

```diff
@@ -0,0 +1,26 @@
+package com.sequenceiq.freeipa.util;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.sequenceiq.cloudbreak.cloud.VersionComparator;
+import com.sequenceiq.cloudbreak.common.type.Versioned;
+import com.sequenceiq.freeipa.entity.Stack;
+
+public class HealthCheckAvailabilityChecker {
+
+    // feature supported from 2.29
+    private static final Versioned CDP_FREEIPA_HEALTH_AGENT_AFTER_VERSION = () -> "2.28.0";
+
+    private HealthCheckAvailabilityChecker() {
+    }
+
+    public static boolean isCdpFreeIpaHeathAgentAvailable(Stack stack) {
```

Comment:
could you drop static and convert this class to `@Component` and use inject in `ClusterProxyService` so we can mock this class easily if we add unit tests for it?

Reply:
Sure. I also updated the other availability checkers to be components.